src/providers/pkcs11_provider/mod.rs

```diff
@@ -928,11 +963,41 @@ impl Provide for Pkcs11Provider {
         info!("Pkcs11 Provider - Asym Verify");
 
         let key_name = op.key_name;
+        let alg = op.alg;
         let mut hash = op.hash;
         let signature = op.signature;
         let key_triple = KeyTriple::new(app_name, ProviderID::Pkcs11, key_name);
         let store_handle = self.key_id_store.read().expect("Key store lock poisoned");
-        let key_id = get_key_id(&key_triple, &*store_handle)?;
+        let (key_id, key_attributes) = get_key_id(&key_triple, &*store_handle)?;
+
+        if !key_attributes.can_verify_hash() {
+            error!("Key {} does not permit verifying hashes.", key_triple);
+            return Err(ResponseStatus::PsaErrorNotPermitted);
+        }
```

Comment:
Do you reckon this could be pushed inside the `can_verify_hash` so it returns a `Result<()>` instead of a `bool`? The error code and log message seem general enough, you could then just do `key_attributes.can_verify_hash()?`.

Same for the `permits_alg` and `is_compatible_with_alg`

Reply:
I think it is nice and clear for those methods to return `bool`, making them return a specific value of `ResponseStatus` feels a bit weird, just to allow this particular code to be simpler.
I could put the log message inside the method but I am not sure it would still be an `error!`: how to be sure that the caller calls it in a context where it is an error? They could intentionally expect the method to return false in a valid case.

Reply:
Will add twin methods for those ones in the interface that return a `Result` and contain the log message.

Reply:
Created parallaxsecond/parsec-interface-rs#39 for that